### PR TITLE
use variable.array instead of variable.data in reporter.py

### PR DESCRIPTION
**Pull request by crcrpar**
This PR is trivial, but from v3, using variable.data is discouraged as noted [here](https://github.com/chainer/chainer/blob/master/chainer/variable.py#L624).

### CODE REVIEW

File: chainer/reporter.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def __call__(self, x, y):
                   # This chain just computes the mean absolute and squared
                   # errors between the prediction and y.
                   pred = self.predictor(x)
-                  abs_error = F.sum(F.abs(pred - y)) / len(x.data)
+                  abs_error = F.sum(F.abs(pred - y)) / len(x.array)
```
Comment (okuta) [inline chainer/reporter.py]: Please use `len(x)` .

Comment (crcrpar) [reply chainer/reporter.py]: OK, thank you.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
Thanks!
LGTM

**Pull request merged**